src/macad_gym/carla/multi_env.py

```diff
@@ -401,13 +406,18 @@ def _init_server(self):
         # Create a new server process and start the client.
         # First find a port that is free and then use it in order to avoid
         # crashes due to:"...bind:Address already in use"
-        self._server_port = MultiCarlaEnv._get_free_tcp_port()
+        self._server_port = self._get_tcp_port()
 
         multigpu_success = False
         gpus = GPUtil.getGPUs()
         log_file = os.path.join(LOG_DIR,
                                 "server_" + str(self._server_port) + ".log")
 
+        print("""Details:
+              1. Port: {}
+              2. Map: {}
+              3. Binary: {}""".format(self._server_port, self._server_map, SERVER_BINARY))
+
```

Comment:
(optional) Suggestion to use `logger.info(...)` instead of `print(...)`.
Also, since the python version in the spec is >3.6, consider using [f-strings](https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python) to make this more compact.

Example: `logger.info(f" 1. Port: {self._server_port} \n 2. Map: {self._server_map} \n 3. Binary: {SERVER_BINARY}")`.
(I used `\n` for the new line character instead of a pre-formatted multi-line string)